### ISSUE TITLE
VS Code doesn't warn if invalid JSON settings are used

### ISSUE THREAD

**Issue by janpb** [valentjn/vscode-ltex#381]
Locally installed ltex-ls is not working in VSCode.

**Expecting** : ltex-ls indicate typos etc. 
**Observing**: No indication of typos

 After opening any LaTeX file, ltex-ls reports following error:
```
The following exception occurred:
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request ltex/workspaceSpecificConfiguration failed with message: s[e] is not iterable
[Info  - 4:52:35 PM] Starting ltex-ls...
Jul 22, 2021 4:52:37 PM org.bsplines.ltexls.server.LtexLanguageServer initialize
INFO: ltex-ls 12.3.0 - initializing...
Jul 22, 2021 4:52:37 PM org.bsplines.ltexls.server.LtexLanguageServer initialize
INFO: Setting locale to 'en'
Jul 22, 2021 4:52:38 PM org.bsplines.ltexls.server.LtexTextDocumentService lambda$didOpen$0
WARNING: The following exception occurred:
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request ltex/workspaceSpecificConfiguration failed with message: s[e] is not iterable
at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
at org.bsplines.ltexls.server.LtexTextDocumentItem.check(LtexTextDocumentItem.java:449)
at org.bsplines.ltexls.server.LtexTextDocumentItem.checkAndGetDiagnostics(LtexTextDocumentItem.java:342)
at org.bsplines.ltexls.server.LtexTextDocumentItem.checkAndPublishDiagnostics(LtexTextDocumentItem.java:319)
at org.bsplines.ltexls.server.LtexTextDocumentItem.checkAndPublishDiagnosticsWithoutCache(LtexTextDocumentItem.java:308)
at org.bsplines.ltexls.server.LtexTextDocumentService.lambda$didOpen$0(LtexTextDocumentService.java:160)
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request ltex/workspaceSpecificConfiguration failed with message: s[e] is not iterable
at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	... 6 more
```

The ltex client is reporting everything is working fine (see below), but  not indicating any typos or other language related issues. The examples in walkthrough  from vscode-ltex (https://github.com/valentjn/vscode-ltex/tree/develop/walkthrough/en) do not indicate expected typos.

Same result using different Java versions (8,10,11), including AdoptOpenJDK (unsetting Java Path).

ltex-client  log:
```
2021-07-22T14:34:16.214Z Info: Setting LTeX UI language to 'en-us'.
2021-07-22T14:34:16.215Z Info: Loading i18n messages...
2021-07-22T14:34:16.215Z Info: Loading default i18n messages...
2021-07-22T14:34:16.218Z Info: 
2021-07-22T14:34:16.218Z Info: ltex.ltex-ls.path set to '/opt/ltex-ls'.
2021-07-22T14:34:16.218Z Info: 
2021-07-22T14:34:16.218Z Info: ltex.java.path set to '/usr/lib/jvm/default/'.
2021-07-22T14:34:16.218Z Info: Using ltex-ls from '/opt/ltex-ls'.
2021-07-22T14:34:16.218Z Info: Using Java from '/usr/lib/jvm/default/'.
2021-07-22T14:34:16.229Z Info: Testing ltex-ls...
2021-07-22T14:34:16.229Z Info:   Command: "/opt/ltex-ls/bin/ltex-ls"
2021-07-22T14:34:16.229Z Info:   Arguments: ["--version"]
2021-07-22T14:34:16.229Z Info:   env['JAVA_HOME']: "/usr/lib/jvm/default/"
2021-07-22T14:34:16.229Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2021-07-22T14:34:16.624Z Info: Test successful!
2021-07-22T14:34:16.626Z Info: 
2021-07-22T14:34:16.632Z Info: Starting ltex-ls...
2021-07-22T14:34:16.632Z Info:   Command: "/opt/ltex-ls/bin/ltex-ls"
2021-07-22T14:34:16.632Z Info:   Arguments: []
2021-07-22T14:34:16.632Z Info:   env['JAVA_HOME']: "/usr/lib/jvm/default/"
2021-07-22T14:34:16.632Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2021-07-22T14:34:16.632Z Info: 
```
ltex-ls version:

```
ltex-ls -V
{
  "ltex-ls": "12.3.0",
  "java": "11.0.11"
}
```
VSCode:
```
Version: 1.58.0
Commit: 2d23c42a936db1c7b3b06f918cde29561cc47cd6
Date: 2021-07-12T23:51:36.303Z
Electron: 12.0.15
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Linux x64 5.13.4-arch1-1
```

**Comment by valentjn** [comment]
Your configuration of `ltex.dictionary` is wrong:

```json
{
  "dictionary": {
    "en-US": [
      "adaptivity",
      "precomputed",
      "subproblem",
      ":~/.config/vscode/ltex-dict/ltex.dictionary.en-US.txt"
    ],
    "de-CH": [
      "B-Splines",
      ":~/.config/vscode/ltex-dict/ltex.dictionary.de-CH.txt"
    ],
    "ltex.disabledRules": {
      "en-US": [
        ":~/.config/vscode/ltex.disabledRules.en-US.txt"
      ],
      "de-CH": [
        ":~/.config/vscode/ltex.disabledRules.de-CH.txt"
      ]
    }
  }
}
```

The key `"ltex.disabledRules"` is neither a valid language, nor is its value an array [as required](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexdictionary).

I was inclined to close this as not-a-bug, but indeed, there was a bug on LT<sub>E</sub>X side, because VS Code is supposed to warn in the `settings.json` file for these kinds of things. Due to missing declarations in LT<sub>E</sub>X's `package.json`, only the values were validated (for valid keys), but it wasn't checked if the keys were actually valid. This is fixed now. Thanks for the report.

**Comment by janpb** [comment]
Everything works after fixing `settings.json`. Should have noticed the wrongly nested `ltex.disabledRules`. Thanks.

**Comment by valentjn** [comment]
`package.json`  fixed in 11.0.0.
